thriftbp/server.go

```diff
@@ -29,19 +38,41 @@ type ServerConfig struct {
 	// which would be too spammy for sentry.
 	Logger thrift.Logger
 
-	// Optional TServerSocket you can use instead of setting one up using Addr
-	// plus timeout.  If provided, this will be used rather than Addr and Timeout.
+	// Optional, used only by NewBaseplateServer.
+	//
+	// Please refer to the documentation of
+	// DefaultProcessorMiddlewaresArgs.ErrorSpanSuppressor for more details
+	// regarding how it is used.
+	ErrorSpanSuppressor errorsbp.Suppressor
+
+	// Optional, used only by NewServer.
```

Comment:
I don't think we really want to say `used only by NewServer`? I think that's confusing since it's still used by `NewBaseplateServer` because that calls `NewServer`.

Reply:
I agree that it is _technically_ correct but I think it's misleading and can cause confusion.

Reply:
How about now?

Reply:
1. `NewBaseplateServer` uses `NewServer` so the field is _still_ used by `NewBaseplateServer`.
2. I don't think it's important to people trying to read the documentation that `NewBaseplateServer` calls `NewServer` and if that ever changes, we'll likely have to use it directly in `NewBaseplateServer`.  From the perspective of someone trying to integrate with this, it _is_ used by `NewBaseplateServer` because that's all they interact with.

Reply:
Ok, I think this works well 🐟